### ISSUE TITLE
Labeled element should not match :hover or :active when the label's activation behavior is "do nothing"

### ISSUE THREAD

**Issue by travisleithead** [w3c/html#293]
Moved from Bugzilla: https://www.w3.org/Bugs/Public/show_bug.cgi?id=28700
@arai-a

> Created [attachment](https://www.w3.org/Bugs/Public/attachment.cgi?id=1602); demo for labeled target with :hover and :active
> 
> Originally reported here:
> - https://bugzilla.mozilla.org/show_bug.cgi?id=1167816
> - https://bugzilla.mozilla.org/show_bug.cgi?id=1167862
> 
> In the attached demo, what the spec says is that hovering/clicking the "Click here" should change the checkbox's style (:hover and :active), but the checkbox shouldn't be checked/unchecked by clicking it.  This is confusing.
> 
> http://w3c.github.io/html/matching-html-elements-using-selectors.html#selectordef-active
> 
> > If the element is the labeled control of a label element that is currently matching :active
> >   The element is being activated.
> 
> Labeled control should be activated only if the label's activation behavior is not "do nothing", since the element won't be actually activated on click.
> 
> http://w3c.github.io/html/matching-html-elements-using-selectors.html#selectordef-hover
> 
> > An element that is the labeled control of a label element that is currently
> > matching :hover.
> 
> This should also be restricted to 'only if the activation behavior is not "do nothing"'.
> 
> > Consider in particular a fragment such as:
> > 
> > ``` html
> >  <p> <label for=c> <input id=a> </label> <span id=b> <input id=c> </span> </p>
> > ```
> > 
> > If the user designates the element with ID "a" with their pointing device,
> > then the p element (and all its ancestors not shown in the snippet above),
> > the label element, the element with ID "a", and the element with ID "c" will
> > match the :hover pseudo-class. The element with ID "a" matches it from
> > condition 1, the label and p elements match it because of condition 2 (one of
> > their descendants is designated), and the element with ID "c" matches it
> > through condition 3 (its label element matches :hover). However, the element
> > with ID "b" does not match :hover: its descendant is not designated, even
> > though it matches :hover.
> 
> Here, ID "c" shouldn't match :hover.

**Comment by siusin** [comment]
We're closing this issue on the W3C HTML specification because the W3C and WHATWG are now [working together on HTML](https://www.w3.org/blog/2019/05/w3c-and-whatwg-to-work-together-to-advance-the-open-web-platform/), and all issues are being discussed on the [WHATWG repository](https://github.com/whatwg/html).

If you filed this issue and you still think it is relevant, please open a [new issue](https://github.com/whatwg/html/issues) on the WHATWG repository and reference this issue (if there is useful information here). Before you open a new issue, please check for existing issues on the WHATWG repository to avoid duplication.

If you have questions about this, please open an issue on the [W3C HTML WG repository](https://github.com/w3c/htmlwg/issues/new) or send an email to public-html@w3.org.
